### ISSUE TITLE
IE 11 Locale 'generic' is not well-formed

### ISSUE THREAD

**Issue by ntdog22** [stephanrauh/ngx-extended-pdf-viewer#357]
**Describe the bug**
When using the ngx-extended-pdf-viewer in Internet Explorer 11, an error is logged to the console:

```
SCRIPT5121: Locale 'generic' is not well-formed 
viewer-es5.min.js (21,124384)
```

This error is _**not**_ raised on any other browser. 

**Version info**
- ngx-extended-pdf-viewer version: **4.0.0-alpha.4**

**Desktop (please complete the following information):**
 - **Internet Explorer 11 running on Windows 10**

**To Reproduce**

1. Go to https://pdfviewer.net/simple in Internet Explorer 11
2. Open up dev tools to see the error logged to the console (might need to reload the page after dev tools are opened).

**Screenshots**
The following screenshot displays the error logged to the console:

![image](https://user-images.githubusercontent.com/6867405/84964526-30ac5b00-b0c1-11ea-9951-64cc505af44c.png)

**Additional context**
***Test Application Information***
- Angular 9.1.11
- Conditional Polyfills configured for IE 11 support

**Comment by stephanrauh** [comment]
Consider it fixed. I've forgotten to add a try-catch-statement. But I don't want to implement this quick-and-dirty solution before examining if there are alternatives. My quick research didn't reveal much, so this sounds like a week-end task.

@ntdog22 Do you happen to know if there are alternatives to using 'generic' on IE11? Such as using the browser locale?

Best regards,
Stephan

**Comment by ntdog22** [comment]
@stephanrauh Thank you for the quick response!

After doing some digging and testing out a few options, I found a some _potential_ solutions for IE 11. I hope this helps!

## Potential Solution 1 ##
**"und"** as the locale value:
From [Unicode's BCP 47 Conformance Doc](https://www.unicode.org/reports/tr35/tr35.html#BCP_47_Conformance), the **und** subtag was found from following backwards compatibility syntax statement:

> The subtag "root" to indicate the generic locale used as the parent of all languages in the CLDR data model ("und" can be used instead)

**Baseline** control case - passing **"en-US"** as the locale string:

![image](https://user-images.githubusercontent.com/6867405/85144323-debe1f00-b1ff-11ea-8ed2-b9235b699f1e.png)

When trying **"root"** in an IE 11 console, the following (similar) error is logged:

![image](https://user-images.githubusercontent.com/6867405/85144145-9acb1a00-b1ff-11ea-8f83-b7e35c5d68df.png)

When trying **"und"**, _no errors_ are logged:

![image](https://user-images.githubusercontent.com/6867405/85144179-aa4a6300-b1ff-11ea-88a3-e5d4092780f9.png)

**"undefined"** seems to work as well?

![image](https://user-images.githubusercontent.com/6867405/85146551-8a686e80-b202-11ea-8681-e327f0d562f3.png)

## Potential Solution 2 ##
**`navigator.x`** as the locale value (I know there are various concerns about this approach):

When using a _simple_ test via IE 11 dev tools:

![image](https://user-images.githubusercontent.com/6867405/85145101-e9c57f00-b200-11ea-8f8f-62f12c860adc.png)

Potential code to circumvent issues with `navigator.x` (from https://github.com/alailsonribeiro/locales/tree/master/language):

```JavaScript

var lg  = {
  a: navigator.language,
  b: navigator.languages,
  c: navigator.userLanguage,
  d: navigator.browserLanguage,
  e: navigator.systemLanguage
};

var lang = {
    detected: lg.a || lg.c || lg.d || lg.e || lg.b[0],
    available: lg.b || [lg.a] || [lg.c] || [lg.d] || [lg.e]
};

```

Screenshot in-action via IE 11:

![image](https://user-images.githubusercontent.com/6867405/85146157-5b51fd00-b202-11ea-8e4f-1761d7174077.png)

## Potential Solution 3 ##
The `try-catch` block as you mentioned.

**Comment by stephanrauh** [comment]
Whow, you've dug really deep! I'll start with the "und" approach. However, I'm not really sure if that's what I want. The Levenstein algorithm is used for fuzzy search - and that means I want o, ó, ö and maybe even ø and œ to be the same letter. If I've got it right, "generic" and "root" distinguish between these letters. But it's probably still better than using the navigator language, which is if the user reads a - say - Spanish page in an English browser installation.

**Comment by stephanrauh** [comment]
The bugfix has landed with version 4.0.0-alpha.5 using your first solution. Thanks for your help!
